From 261fc47ebcd7bf54334d333fc6fbbe04b850c51f Mon Sep 17 00:00:00 2001
From: Ihor Usyk <ihor_usyk@epam.com>
Date: Tue, 2 May 2023 13:45:58 +0200
Subject: [PATCH 1/3] doma, domd: decompose the device tree to support second
 hdmi

Doma and Domd device trees now have been separated:
DomD
r8a77951-h3ulcb-4x2g-kf-hdmi-domd.dts - specify HDMI display
r8a77951-h3ulcb-4x2g-kf-lvds-domd.dts - specify LVDS display

both files include r8a77951-h3ulcb-4x2g-kf-domd.dtsi, the file
describe the common configuration of DomD

DomA
r8a7795-h3ulcb-4x2g-kf-hdmi-doma.dts - specify HDMI display
r8a7795-h3ulcb-4x2g-kf-lvds-doma.dts - specify LVDS display

both files include r8a7795-h3ulcb-4x2g-kf-doma.dtsi, the file
describe the common configuration of DomA

The new moulin parameter has been introduced to configure the required
display:

DISPLAY {"lvds", "hdmi"}
"lvds" (default value}
Add the files r8a77951-h3ulcb-4x2g-kf-lvds-domd.dts and r8a7795-h3ulcb-4x2g-kf-lvds-doma.dts
to build.

"hdmi"
Add the files r8a77951-h3ulcb-4x2g-kf-hdmi-domd.dts and r8a7795-h3ulcb-4x2g-kf-hdmi-doma.dts
to build.

Signed-off-by: Ihor Usyk <ihor_usyk@epam.com>
---
 ...a.dts => r8a7795-h3ulcb-4x2g-kf-doma.dtsi} |  2 +-
 .../r8a7795-h3ulcb-4x2g-kf-hdmi-doma.dts      | 14 +++++++++++++
 .../r8a7795-h3ulcb-4x2g-kf-lvds-doma.dts      | 14 +++++++++++++
 ....dts => r8a77951-h3ulcb-4x2g-kf-domd.dtsi} | 13 ++++++++++--
 .../r8a77951-h3ulcb-4x2g-kf-hdmi-domd.dts     | 13 ++++++++++++
 .../r8a77951-h3ulcb-4x2g-kf-lvds-domd.dts     | 13 ++++++++++++
 .../linux/linux-renesas_%.bbappend            |  2 ++
 prod-cockpit-rcar.yaml                        | 21 ++++++++++++++++---
 8 files changed, 86 insertions(+), 6 deletions(-)
 rename meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/{r8a7795-h3ulcb-4x2g-kf-doma.dts => r8a7795-h3ulcb-4x2g-kf-doma.dtsi} (99%)
 create mode 100644 meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-hdmi-doma.dts
 create mode 100644 meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-lvds-doma.dts
 rename meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/{r8a77951-h3ulcb-4x2g-kf-domd.dts => r8a77951-h3ulcb-4x2g-kf-domd.dtsi} (97%)
 create mode 100644 meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-hdmi-domd.dts
 create mode 100644 meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-lvds-domd.dts

diff --git a/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-doma.dts b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-doma.dtsi
similarity index 99%
rename from meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-doma.dts
rename to meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-doma.dtsi
index 9684ec5..7be0fab 100644
--- a/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-doma.dts
+++ b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-doma.dtsi
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: GPL-2.0
 /dts-v1/;
 
 #include <dt-bindings/interrupt-controller/arm-gic.h>
@@ -68,7 +69,6 @@
                                 };
                         };
 			display-1 {
-				display-map = <0>;
 				layers {
 					primary {
 						layer-map=<0x4>;
diff --git a/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-hdmi-doma.dts b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-hdmi-doma.dts
new file mode 100644
index 0000000..861c787
--- /dev/null
+++ b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-hdmi-doma.dts
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-2.0
+/dts-v1/;
+
+#include "r8a7795-h3ulcb-4x2g-kf-doma.dtsi"
+
+/ {
+	rvgc {
+		displays {
+			display-1 {
+				display-map = <0x3>;
+			};
+		};
+	};
+};
diff --git a/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-lvds-doma.dts b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-lvds-doma.dts
new file mode 100644
index 0000000..327a3e4
--- /dev/null
+++ b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a7795-h3ulcb-4x2g-kf-lvds-doma.dts
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-2.0
+/dts-v1/;
+
+#include "r8a7795-h3ulcb-4x2g-kf-doma.dtsi"
+
+/ {
+	rvgc {
+		displays {
+			display-1 {
+				display-map = <0x0>;
+			};
+		};
+	};
+};
diff --git a/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-domd.dts b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-domd.dtsi
similarity index 97%
rename from meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-domd.dts
rename to meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-domd.dtsi
index 7c8da14..214f292 100644
--- a/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-domd.dts
+++ b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-domd.dtsi
@@ -44,7 +44,6 @@
                 };
                 displays {
                         display-0 {
-                                display-map = <0>;
                                 layers {
                                         primary{
                                                 /* Layer shold display the information at all display */
@@ -109,6 +108,7 @@
 	/delete-node/ cvbs-in;
 	/delete-node/ hdmi-in;
 	/delete-node/ hdmi0-out;
+	/delete-node/ hdmi-out;
 	/delete-node/ lvds-decoder;
 
 	/delete-node/ lvds;
@@ -116,6 +116,16 @@
 	/delete-node/ vga-encoder;
 };
 
+&i2c2 {
+       i2c-switch@71 {
+               /delete-node/i2c@4;
+       };
+};
+
+&pfc {
+       /delete-node/ du;
+};
+
 /delete-node/&adsp;
 
 /* Xen will provide its own GIC, mask ours */
@@ -155,7 +165,6 @@
 /delete-node/ &vspd2;
 /delete-node/ &du;
 /delete-node/ &hdmi0;
-/delete-node/ &adv7513_in;
 /delete-node/ &hdmi1;
 /delete-node/ &lvds0;
 /delete-node/ &pciec0;
diff --git a/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-hdmi-domd.dts b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-hdmi-domd.dts
new file mode 100644
index 0000000..4f00d62
--- /dev/null
+++ b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-hdmi-domd.dts
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include "r8a77951-h3ulcb-4x2g-kf-domd.dtsi"
+
+/ {
+	rvgc {
+		displays {
+			display-0 {
+				display-map = <3>;
+			};
+		};
+	};
+};
diff --git a/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-lvds-domd.dts b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-lvds-domd.dts
new file mode 100644
index 0000000..1e42255
--- /dev/null
+++ b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-kf-lvds-domd.dts
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include "r8a77951-h3ulcb-4x2g-kf-domd.dtsi"
+
+/ {
+	rvgc {
+		displays {
+			display-0 {
+				display-map = <0>;
+			};
+		};
+	};
+};
diff --git a/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/linux-renesas_%.bbappend b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/linux-renesas_%.bbappend
index 1021f41..85b4a73 100644
--- a/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/linux-renesas_%.bbappend
+++ b/meta-xt-prod-cockpit-rcar-driver-domain/recipes-kernel/linux/linux-renesas_%.bbappend
@@ -8,5 +8,7 @@ SRC_URI_append = " \
     file://defconfig \
     file://xt_pass_drv.cfg \
     file://boot-options.dtsi;subdir=git/arch/${ARCH}/boot/dts/renesas \
+    file://r8a7795-h3ulcb-4x2g-kf-doma.dtsi;subdir=git/arch/${ARCH}/boot/dts/renesas \
+    file://r8a77951-h3ulcb-4x2g-kf-domd.dtsi;subdir=git/arch/${ARCH}/boot/dts/renesas \
 "
 
diff --git a/prod-cockpit-rcar.yaml b/prod-cockpit-rcar.yaml
index 47d1995..e8afb3a 100644
--- a/prod-cockpit-rcar.yaml
+++ b/prod-cockpit-rcar.yaml
@@ -8,7 +8,7 @@ variables:
   CLUSTER_BUILD_DIR: "build-cluster-wrapper"
   TOOLS_BUILD_DIR: "build-tools-native"
   ANDROID_KERNEL_DIR: "android_kernel"
-  XT_DOMD_DTB_NAME: "%{SOC_FAMILY}-%{MACHINE}-domd.dtb"
+  XT_DOMD_DTB_NAME: "%{SOC_FAMILY}-%{MACHINE}-lvds-domd.dtb"
   XT_XEN_DTB_NAME: "%{SOC_FAMILY}-%{MACHINE}-xen.dtb"
   XT_XEN_REVISION: "995f0fcf697d8d8690ed02e0990b2be074ebe8cb"
   XT_PVR_NUM_OSID: "2"
@@ -391,10 +391,10 @@ parameters:
           SOC_FAMILY: "r8a7795"
           XT_DOMD_CONFIG_NAME: "domd-h3ulcb-4x2g-kf.cfg"
           XT_OP_TEE_FLAVOUR: "salvator_h3_4x2g"
-          XT_DOMD_DTB_NAME: "r8a77951-h3ulcb-4x2g-kf-domd.dtb"
+          XT_DOMD_DTB_NAME: "r8a77951-h3ulcb-4x2g-kf-lvds-domd.dtb"
           XT_XEN_DTB_NAME: "r8a77951-h3ulcb-4x2g-kf-xen.dtb"
           XT_DOMA_CONFIG_NAME: "doma-h3ulcb-4x2g-kf.cfg"
-          XT_DOMA_DTB_NAME: "r8a7795-h3ulcb-4x2g-kf-doma.dtb"
+          XT_DOMA_DTB_NAME: "r8a7795-h3ulcb-4x2g-kf-lvds-doma.dtb"
         components:
           domd:
             sources:
@@ -528,3 +528,18 @@ parameters:
             builder:
               additional_deps:
                 - "%{TOOLS_BUILD_DIR}/tmp/deploy/cr7-image-gen"
+
+
+  DISPLAY:
+    desc: "Build for HDMI display, default is LVDS"
+    "lvds":
+      default: true
+      overrides:
+        variables:
+          XT_DOMD_DTB_NAME: "r8a77951-h3ulcb-4x2g-kf-lvds-domd.dtb"
+          XT_DOMA_DTB_NAME: "r8a7795-h3ulcb-4x2g-kf-lvds-doma.dtb"
+    "hdmi":
+      overrides:
+        variables:
+          XT_DOMD_DTB_NAME: "r8a77951-h3ulcb-4x2g-kf-hdmi-domd.dtb"
+          XT_DOMA_DTB_NAME: "r8a7795-h3ulcb-4x2g-kf-hdmi-doma.dtb"
-- 
2.25.1

